eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -41,15 +41,15 @@ jobs:
       condition: and(succeeded(), eq(dependencies.smoke_test_eligibility.outputs['output_eligibility.RunSmokeTests'], true))
     strategy:
       matrix:
-        Python_38_Linux (AzureCloud):
-          PythonVersion: '3.8'
-          Pool: "azsdk-pool-mms-ubuntu-2004-general"
-          OSVmImage: azsdk-pool-mms-ubuntu-2004-1espt
+        Python_310_Linux (AzureCloud):
```

Comment:
@danieljurek see for example here - working matrix vars using the image.yml template. Re: the cpp changes.